src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp

```diff
@@ -54,11 +66,63 @@ struct Atomic::PlatformAdd
   }
 };
 
+#ifndef FULL_COMPILER_ATOMIC_SUPPORT
+template<>
+template<typename T>
+inline T Atomic::PlatformCmpxchg<1>::operator()(T volatile* dest __attribute__((unused)),
+                                                T compare_value,
+                                                T exchange_value,
+                                                atomic_memory_order order) const {
```

Comment:
I see the parameter order is not consistent with the one in jdk11u:
`T Atomic::PlatformCmpxchg<byte_size>::operator()`.
So this might need more adaptations, otherwise I guess it won't build.
Also the original commit adds one `#undef FULL_COMPILER_ATOMIC_SUPPORT` at the end of the file, which is missing for this backport.

Reply:
It did build and run... I'll fix thanks.

Reply:
I probably used to new compiler...